Lib/unittest/mock.py

```diff
@@ -885,7 +906,15 @@ def _get_child_mock(self, **kw):
 
         For non-callable mocks the callable variant will be used (rather than
         any custom subclass)."""
+        _new_name = kw.get("_new_name")
+        if _new_name in self.__dict__['_spec_asyncs']:
+            return AsyncMock(**kw)
+
         _type = type(self)
+        if issubclass(_type, MagicMock) and _new_name in _async_method_magics:
+            klass = AsyncMock
+        if issubclass(_type, AsyncMockMixin):
+            klass = MagicMock
```

Comment:
How much of all this kind of thing could be avoided if the docs just said "if you want to use async stuff, even to spec, just use `AsyncMock` from the start?"

Reply:
Not a large amount, I think 9 lines of code could be avoided. I don't think it would be worth it to remove the functionality.

Reply:
It would mean speccing an object that has async attributes/methods etc wouldn't produce async compatible mocks where it should.